src/ui/public/selectable_row/selectable_row.js

```diff
@@ -0,0 +1,22 @@
+import { uiModules } from 'ui/modules';
+import template from './selectable_row.html';
+
+const app = uiModules.get('kibana');
+
+app.directive('selectableRow', function () {
+  return {
+    restrict: 'E',
+    replace: true,
+    transclude: true,
+    template: template,
+    scope: {
+      id: '=',
+      isSelected: '=',
+      onSelectChange: '=',
+    },
+    controllerAs: 'selectableRow',
```

Comment:
Thanks for the discussion, folks. I'm going with option 1, which means keeping the code in this PR as-is.